### PR TITLE
todos

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: tests/Runner/RunnerTest.php [hunk 1]
```diff
@@ -184,7 +184,7 @@ public function testThatParallelFixOfInvalidFileReportsToErrorManager(): void
     /**
      * @requires OS Darwin|Windows
      *
-     * @TODO Fix it for Linux - somehow this test reaches timeout in CI, even though it works on Linux through Docker.
+     * @TODO v4 Fix it for Linux - somehow this test reaches timeout in CI, even though it works on Linux through Docker. @Wirone
```
Comment (Wirone) [inline tests/Runner/RunnerTest.php:187]: I don't think we need to wait for v4 to fix it. I have a suspicion what's wrong here - probably it's a race condition between spawning child processes and stopping all of them after encountering violation. This error looks like 1st process fails on violation and kills all processes before 2nd process' `hello` action is handled, so `$processPool->getProcess()` fails with exception. Maybe timeout is somehow similar, but 2nd process is stopped after process is fetched from the pool, but connection can't be established? I would like to look at this, but I don't consider it as a blocker.

File: src/Error/ErrorsManager.php [hunk 2]
```diff
@@ -96,6 +96,8 @@ public function report(Error $error): void
         $this->errors[] = $error;
     }
 
+    // @TODO v4 move handling of WorkerException outside of ErrorsManager and break the execution flow when it happen @Wirone
+    // ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777#discussion_r1591666901
```
Comment (Wirone) [inline src/Error/ErrorsManager.php:100]: Not needed, as error handling has been [reworked](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777#issuecomment-2109111690).

File: src/Console/Command/WorkerCommand.php [hunk 3]
```diff
@@ -182,6 +182,8 @@ function (ConnectionInterface $connection) use ($loop, $runner, $identifier): vo
                     });
                 },
                 static function (\Throwable $error) use ($errorOutput): void {
+                    // @TODO v4 verify onRejected behaviour @Wirone
+                    // ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777#discussion_r1590399285
```
Comment (Wirone) [inline src/Console/Command/WorkerCommand.php:186]: Added during [error handling rework](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777#issuecomment-2109111690).

File: tests/Runner/RunnerTest.php (hunk 1 above)
Comment (keradus) [reply tests/Runner/RunnerTest.php:187]: > I don't think we need to wait for v4 to fix it

my point was not to wait for v4 indeed, but to not release v4 with parallel execution by default if this is not yet solved.

happy to have a v4 mentioned again, with context of above.
I use `grep TODO v4` as quick filter.

**Pull request closed without merging**